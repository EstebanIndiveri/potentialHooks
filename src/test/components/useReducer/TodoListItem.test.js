import React from 'react';
import { shallow } from 'enzyme';
import TodoListItem from '../../../components/useReducer/TodoListItem';
import { demoTodos } from '../../fixtures/demoTodos';
import '@testing-library/jest-dom';

describe('<TodoListItem/>', () => {
    const handleDelete=jest.fn();
    const handleToggle=jest.fn();
    const wrapper=shallow(<TodoListItem 
        todo={demoTodos[0]} 
        index={demoTodos[0].id} 
        handleDelete={handleDelete} 
        handleToggle={handleToggle}
        />)
    test('debe de mostrar correctamente', () => {
        //snap
        expect(wrapper).toMatchSnapshot();
    });

    test('llama function handleDelete', () => {
        //jest.fn();
        //havebeencalledwith
        wrapper.find('button').simulate('click');
        expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
    });

    test('llama functon handleToggle', () => {
        //jest.fn()
        //havebeencalledwith
        wrapper.find('p').simulate('click');
        expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
    });
    test('mostrar el texto correctamente', () => {
        //p content
        const p=wrapper.find('p');
        expect(p.text().trim()).toBe(`${demoTodos[0].id + 1 }- ${demoTodos[0].desc}`)
    })
    

})
