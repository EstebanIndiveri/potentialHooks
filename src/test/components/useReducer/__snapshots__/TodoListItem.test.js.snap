// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListItem/> debe de mostrar correctamente 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    2
    - 
    aprender react
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</li>
`;
